scripts/lc-builds/toss4_amdclang.sh

```diff
@@ -53,23 +59,28 @@ module load cmake/3.23.1
 
 # unload rocm to avoid configuration problems where the loaded rocm and COMP_VER
 # are inconsistent causing the rocprim from the module to be used unexpectedly
-module unload rocm
+# module unload rocm
 
+if [[ ${COMP_VER} =~ .*magic.* ]]; then
+  ROCM_PATH="/usr/tce/packages/rocmcc/rocmcc-${COMP_VER}"
+else
+  ROCM_PATH="/usr/tce/packages/rocmcc-tce/rocmcc-${COMP_VER}"
+fi
 
 cmake \
   -DCMAKE_BUILD_TYPE=Release \
-  -DROCM_ROOT_DIR="/opt/rocm-${COMP_VER}" \
-  -DHIP_ROOT_DIR="/opt/rocm-${COMP_VER}/hip" \
-  -DHIP_PATH=/opt/rocm-${COMP_VER}/llvm/bin \
-  -DCMAKE_C_COMPILER=/opt/rocm-${COMP_VER}/llvm/bin/amdclang \
-  -DCMAKE_CXX_COMPILER=/opt/rocm-${COMP_VER}/llvm/bin/amdclang++ \
-  -DCMAKE_HIP_ARCHITECTURES="${COMP_ARCH}" \
-  -DGPU_TARGETS="${COMP_ARCH}" \
-  -DAMDGPU_TARGETS="${COMP_ARCH}" \
+  -DROCM_ROOT_DIR="${ROCM_PATH}" \
+  -DHIP_ROOT_DIR="${ROCM_PATH}/hip" \
+  -DHIP_PATH=${ROCM_PATH}/llvm/bin \
+  -DCMAKE_C_COMPILER=${ROCM_PATH}/llvm/bin/amdclang \
+  -DCMAKE_CXX_COMPILER=${ROCM_PATH}/llvm/bin/amdclang++ \
+  -DCMAKE_HIP_ARCHITECTURES="${COMP_ARCH}:xnack+" \
+  -DGPU_TARGETS="${COMP_ARCH}:xnack+" \
+  -DAMDGPU_TARGETS="${COMP_ARCH}:xnack+" \
```

Comment:
Why are we forcing xnack+ here? Are we trying to allow people to use address sanitizer?

Reply:
That's a cut & paste error. Fixed.